controllers/metric_controller.go

```diff
@@ -96,16 +97,22 @@ func (r *metricReconciler) validate(ctx context.Context) (bool, error) {
 		return false, fmt.Errorf("error getting operator namespace: %w", err)
 	}
 	// if the label exists, update value from default of false
-	if value, ok := wmcoNamespace.Labels["openshift.io/cluster-monitoring"]; ok {
+	if value, ok := wmcoNamespace.Labels[monitoringLabel]; ok {
 		labelValue, err = strconv.ParseBool(value)
 		if err != nil {
 			return false, fmt.Errorf("monitoring label must have a boolean value: %w", err)
 		}
 	}
-	if !labelValue {
-		r.recorder.Eventf(wmcoNamespace, v1.EventTypeWarning, "labelValidationFailed",
-			"Cluster monitoring openshift.io/cluster-monitoring=true label is not enabled in %s namespace",
-			r.watchNamespace)
+	if labelValue {
+		// If the label exists, log the updated value
+		r.recorder.Eventf(wmcoNamespace, v1.EventTypeNormal, "monitoringEnabled",
+			"Cluster monitoring %s label is enabled in %s namespace",
+			monitoringLabel, r.watchNamespace)
+	} else {
+		// If the label is removed, log the removal
+		r.recorder.Eventf(wmcoNamespace, v1.EventTypeWarning, "monitoringDisabled",
+			"Cluster monitoring %s label is not enabled in %s namespace",
+			monitoringLabel, r.watchNamespace)
```

Comment:
This is fine, but wouldnt keeping track of the previous state in the metricReconciler object make more sense? Generating an event if the prev state != new state.
This would be more defensive, that way this function doesn't completely rely on the predicate to not generate unnecessary events